src/100-App/redux-store.tsx

```diff
@@ -137,40 +153,85 @@ export const logoutAccount = async(dispatch: (arg0: {type: 'account/resetAccount
+  skipAnimation:boolean,
+  lastNewPartnerRequest:number|undefined, //timestamp
 }
 
 const initialSettingsState:SettingsState = {
+  version: parseInt(process.env.REACT_APP_SETTINGS_VERSION ?? '1', 10),
   ignoreCache: false,
   skipAnimation: false,
-  lastNewPartnerRequest: Date.now() - 60 * 60 * 1000, //1 hour ago
-}; 
+  lastNewPartnerRequest: undefined,
+};
+
+//Use as default; but don't save to local storage
+export const DEFAULT_LAST_NEW_PARTNER_REQUEST:number = Date.now() - parseInt(process.env.REACT_APP_NEW_PARTNER_TIMEOUT ?? '3600000', 10);  //1 hour ago
  
 const settingsSlice = createSlice({
   name: 'settings',
   initialState: initialSettingsState,
   reducers: {
+    setSettings: (state, action:PayloadAction<SettingsState>) => state = {...action.payload},
+    resetSettings: () => initialSettingsState,
     setIgnoreCache: (state, action:PayloadAction<boolean>) => state = {...state, ignoreCache: action.payload},
     setSkipAnimation: (state, action:PayloadAction<boolean>) => state = {...state, skipAnimation: action.payload},
     setLastNewPartnerRequest: (state) => state = {...state, lastNewPartnerRequest: Date.now()},
-    resetLastNewPartnerRequest: (state) => state = {...state, lastNewPartnerRequest: initialSettingsState.lastNewPartnerRequest},
-    resetSettings: () => initialSettingsState,
+    resetLastNewPartnerRequest: (state) => state = {...state, lastNewPartnerRequest: undefined},    
   },
 });
 
 //Export Dispatch Actions
-export const { setIgnoreCache, setSkipAnimation, 
+export const { setSettings, resetSettings,
+    setIgnoreCache, setSkipAnimation, 
     setLastNewPartnerRequest, resetLastNewPartnerRequest,
-    resetSettings } = settingsSlice.actions;
+} = settingsSlice.actions;
+
 
+export const initializeSettingsState = async(dispatch: (arg0: { payload: SettingsState; type: 'settings/setSettings'; }|{type: 'settings/resetSettings'; }) => void, getState: AccountState) => {
+    try {
+        const localStorageSettings:string|null = localStorage.getItem('settings');
+        const savedSettings:SettingsState = localStorageSettings ? JSON.parse(localStorageSettings) : initialSettingsState;
+        if(!isNaN(savedSettings.version) && (savedSettings.version == parseInt(process.env.REACT_APP_SETTINGS_VERSION ?? '1', 10)))
+          dispatch(setSettings({ ...initialSettingsState, ...savedSettings }));
+
+        else
+          throw new Error('Invalid Settings');
+
+    } catch (error) {
+        console.error('REDUX Settings | localStorage initialization failed: ', error);
+        dispatch(resetSettings());
+    }
+};
+
+
+export const saveSettingsMiddleware:Middleware = store => next => action => {
+  const result = next(action);
+
+  if(Object.values(settingsSlice.actions).map(action => action.type).includes(action.type) && action.type !== resetSettings.type) {
+    const settingsState: RootState['settings'] = store.getState().settings;
+    localStorage.setItem('settings', JSON.stringify(settingsState));
+
```

Comment:
The above code is very clean, nice way to dynamically save settings

Reply:
Thanks, middlewear gets called on every dispatch for all reducers.  I was looking for a dynamic way to filter; this should work good.